### PR TITLE
Use TaskclusterConfig from official library

### PR DESCRIPTION
**Pull request by La0**
:warning: some updates will be needed on code-review & code-coverage as they are the ones doing the auth call

### CODE REVIEW

File: libmozevent/monitoring.py [hunk 1]
```diff
@@ -26,11 +25,14 @@ class Monitoring(object):
     every X seconds
     """
 
-    def __init__(self, queue_name, emails, period):
-        assert isinstance(queue_name, str)
-        assert isinstance(period, int)
+    def __init__(
+        self,
+        taskcluster_config: TaskclusterConfig,
```
Comment (marco-c) [inline libmozevent/monitoring.py]: This should fix #17, right?

File: requirements.txt [hunk 2]
```diff
@@ -5,4 +5,4 @@ Logbook
 python-hglib
 pytoml
 structlog
-taskcluster
+taskcluster==24.0.0
```
Comment (marco-c) [inline requirements.txt]: Can we do >=? Or not specify it at all?

File: libmozevent/monitoring.py (hunk 1 above)
Comment (La0) [reply libmozevent/monitoring.py]: I think so

File: requirements.txt (hunk 2 above)
Comment (La0) [reply requirements.txt]: I prefer to explicitly set the min version, will switch to `>=`

**Pull request merged**